### PR TITLE
Qt: Add Classic Windows as theme option

### PR DESCRIPTION
**Pull request by JordanTheToaster**
### Description of Changes
Shamelessly ~~stolen~~ backported from Duckstation.

### Rationale behind Changes
For anyone who doesn't have eyes and also does not like the default Windows 11 theme?

### Suggested Testing Steps
Make sure CI is happy boi.

### CODE REVIEW

File: pcsx2-qt/Themes.cpp [hunk 1]
```diff
@@ -55,6 +55,16 @@ void QtHost::SetStyleFromSettings()
 		qApp->setStyle(QStyleFactory::create("Fusion"));
 		qApp->setStyleSheet(QString());
 	}
+#ifdef _WIN32
+	else if (theme == "windowsvista")
+	{
+		// setPalette() shouldn't be necessary, as the documentation claims that setStyle() resets the palette, but it
+		// is here, to work around a bug in 6.4.x and 6.5.x where the palette doesn't restore after changing themes.
+		qApp->setPalette(QPalette());
```
Comment (stenzek) [inline pcsx2-qt/Themes.cpp:63]: Can you please move this between lines 49 and 50? It's repeated on several of the branches here.

Comment (JordanTheToaster) [reply pcsx2-qt/Themes.cpp]: Done.

**Comment by stenzek** [comment]
_WIN32 is defined on both 32 and 64 bit, and is the "standard" macro for windows.

It has to be ifdef'ed because AFAIK the windowsvista theme does not exist on other OSes.

**Review by kamfretoz — APPROVED** [review]

**Comment by TheLastRar** [comment]
This is probably a QT issue, but the dropdown menus have a rendering glitch on Windows 11
image
They gain a vertical white bar between the icons and text.
That issue persists even when restarting PCSX2 after applying the theme change

Windows 10 doesn't have this issue from what I've been told.

Switching back and forth between classic and native without restarting also really breaks rendering of these menus, but I'm assuming that is also a Qt bug

**Comment by stenzek** [comment]
Yeah, I noticed that as well. May be able to work around the switch issue by recreating the window, but honestly, I'm inclined to just leave it until if/when Qt sort it out upstream. Not like these are the default themes, and they're quite.. ugly anyway.

**Comment by TheLastRar** [comment]
iirc the switch issue affected other themes aswell.

The classic theme, however, remains bugged even after restarting PCSX2, and ends up looking worse on 11 than on 10.

**Pull request merged**